specifyweb/frontend/js_src/lib/components/WorkBench/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { HotTable } from '@handsontable/react';
+
+import { useAsyncState } from '../../hooks/useAsyncState';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { useErrorContext } from '../../hooks/useErrorContext';
+import { commonText } from '../../localization/common';
+import { wbText } from '../../localization/workbench';
+import { ajax } from '../../utils/ajax';
+import { f } from '../../utils/functools';
+import type { GetSet } from '../../utils/types';
+import { className } from '../Atoms/className';
+import { LoadingContext } from '../Core/Contexts';
+import { LoadingScreen } from '../Molecules/Dialog';
+import { useMenuItem } from '../Header/MenuContext';
+import { treeRanksPromise } from '../InitialContext/treeRanks';
+import { Dialog } from '../Molecules/Dialog';
+import { NotFoundView } from '../Router/NotFoundView';
+import type { Dataset } from '../WbPlanView/Wrapped';
+import { WbViewReact } from './WbView';
+
+const fetchTreeRanks = async (): Promise<true> => treeRanksPromise.then(f.true);
+
+const fetchDataSet = async (
+  dataSetId: number | undefined
+): Promise<Dataset | undefined> =>
+  typeof dataSetId === 'number'
+    ? ajax<Dataset>(`/api/workbench/dataset/${dataSetId}/`, {
+        headers: { Accept: 'application/json' },
+      }).then(({ data }) => data)
+    : undefined;
+
+// BUG: intercept 403 (if dataset has been transferred to another user)
+function useDataSet(
+  dataSetId: number | undefined
+): GetSet<Dataset | undefined> {
+  return useAsyncState(
+    React.useCallback(async () => fetchDataSet(dataSetId), [dataSetId]),
+    true
+  );
+}
+
+export function WorkBenchReact(): JSX.Element {
+  useMenuItem('workBench');
+
+  const [treeRanksLoaded = false] = useAsyncState(fetchTreeRanks, true);
+  const { id } = useParams();
+  const dataSetId = f.parseInt(id);
+
+  const [dataSet, setDataSet] = useDataSet(dataSetId);
+  useErrorContext('dataSet', dataSet);
+  const loading = React.useContext(LoadingContext);
+  const [isDeleted, handleDeleted] = useBooleanState();
+  const [isDeletedConfirmation, handleDeletedConfirmation] = useBooleanState();
+
+  const navigate = useNavigate();
+  const hotRef = React.useRef<HotTable>(null);
+  const spreadsheetContainer = React.useRef<HTMLElement | null>(null);
+
+  if (!dataSet || !treeRanksLoaded) return <LoadingScreen />;
+
+  const triggerRefresh = () => {
+    loading(fetchDataSet(dataSet!.id).then(setDataSet));
+  };
```

Comment:
👍

also, if necessary, consider adding a `key={someState}` prop to WbViewReact and incrementing that `someState` on each refresh - this will completely nuke all the states inside of WbViewReact and all the things inside of it. This is needed in case some components still have stale state after this "refresh"